sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -3082,7 +3082,10 @@ object functions {
    * @since 1.5.0
    */
   def array_contains(column: Column, value: Any): Column = withExpr {
-    ArrayContains(column.expr, Literal(value))
+    value match {
```

Comment:
@chongguang I mean, use just `ArrayContains(column.expr, lit(value).expr)` instead of `value match { ...`. The `lit()` should handle literals and columns well.